NgUpgrade AOT bug introduced in 4.0.0-beta.8

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION WE MIGHT CLOSE YOUR ISSUE WITHOUT INVESTIGATING
-->

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Source**
<!-- Describe how the bug manifests. -->

```ts
// Source file
import { UpgradeModule } from '@angular/upgrade/static'
```

**AOT Compiled**
<!-- Describe what the behavior would be without the bug. -->

```ts
// Example AOT compiled - Angular 4.0.0-beta.7
import * as import5 from '@angular/upgrade/src/static/upgrade_module';

// Example AOT compiled - Angular 4.0.0-beta.8
import * as import4 from '../../../@angular/upgrade/static';
```

This naturally results in the following:

> app.module.ngfactory.ts:12:26: Cannot find module '../../../@angular/upgrade/static'.

I can only imagine something about this PR https://github.com/angular/angular/pull/14509 is not quite right...

**What is the motivation / use case for changing the behavior?**
Bug fix
<!-- Describe the motivation or the concrete use case -->

**Please tell us about your environment:** macOS 10.12.3
<!-- Operating system, IDE, package manager, HTTP server, ... -->

* **Angular version:** 4.0.0-beta.8
<!-- Check whether this is still an issue in the most recent Angular version -->
 
* **Language:** TypeScript 2.2.0 (Current RC)

* **Node (for AoT issues):** `node --version` =   6.9.4

--- a/modules/@angular/compiler-cli/src/compiler_host.ts
+++ b/modules/@angular/compiler-cli/src/compiler_host.ts
@@ -18,7 +18,7 @@ const NODE_MODULES = '/node_modules/';
 const IS_GENERATED = /\.(ngfactory|ngstyle)$/;
 const GENERATED_FILES = /\.ngfactory\.ts$|\.ngstyle\.ts$/;
 const GENERATED_OR_DTS_FILES = /\.d\.ts$|\.ngfactory\.ts$|\.ngstyle\.ts$/;
-const SHALLOW_IMPORT = /^((\w|-)+|(@(\w|-)+\/(\w|-)+))$/;
+const SHALLOW_IMPORT = /^((\w|-)+|(@(\w|-)+(\/(\w|-)+)+))$/;
 
 export interface CompilerHostContext extends ts.ModuleResolutionHost {
   readResource(fileName: string): Promise<string>;

--- a/modules/@angular/compiler-cli/test/aot_host_spec.ts
+++ b/modules/@angular/compiler-cli/test/aot_host_spec.ts
@@ -135,6 +135,18 @@ describe('CompilerHost', () => {
         .toEqual('@angular/core');
   });
 
+  it('should be able to produce an import to a shallow import', () => {
+    expect(hostNestedGenDir.fileNameToModuleName('@angular/core', '/tmp/project/src/main.ts'))
+        .toEqual('@angular/core');
+    expect(hostNestedGenDir.fileNameToModuleName(
+               '@angular/upgrade/static', '/tmp/project/src/main.ts'))
+        .toEqual('@angular/upgrade/static');
+    expect(hostNestedGenDir.fileNameToModuleName('myLibrary', '/tmp/project/src/main.ts'))
+        .toEqual('myLibrary');
+    expect(hostNestedGenDir.fileNameToModuleName('lib23-43', '/tmp/project/src/main.ts'))
+        .toEqual('lib23-43');
+  });
+
   it('should be able to produce an import from main to a sub-directory', () => {
     expect(hostNestedGenDir.fileNameToModuleName('lib/utils.ts', 'main.ts')).toEqual('./lib/utils');
   });

